src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,551 @@
+            further processed by a Linear layer. This output is to be used to embed contexts for
+            nearest neighbors queries with question embeddings.
+
+    Examples::
+
+        tokenizer = DprTokenizer.from_pretrained('dpr-base-uncased')
+        model = DprContextEncoder.from_pretrained('dpr-ctx_encoder-base')
+        input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)  # Batch size 1
+        embeddings = model(input_ids)  # the embeddings of the given context.
+
+    """
+
+    def __init__(self, config: DprConfig):
+        super().__init__(config)
+        self.config = config
+        self.ctx_encoder = get_bert_ctx_encoder_component(config)
+        self.init_weights()
+
+    def forward(self, input_ids: T, token_type_ids: Optional[T] = None, attention_mask: Optional[T] = None) -> T:
+        if attention_mask is None:
+            attention_mask = input_ids != self.config.pad_id
+            attention_mask = attention_mask.to(device=input_ids.device)
+        sequence_output, pooled_output, hidden_states = self.ctx_encoder(input_ids, token_type_ids, attention_mask)
+        return pooled_output
+
+
+@add_start_docstrings(
+    "The bare DprQuestionEncoder transformer outputting pooler outputs as question representations.",
+    DPR_START_DOCSTRING,
+    DPR_ENCODERS_INPUTS_DOCSTRING,
+)
+class DprQuestionEncoder(DprPretrainedQuestionEncoder):
+    r"""
+    Outputs: The DPR encoder only outputs the `pooler_output` that corresponds to the question representation:
+        **pooler_output**: ``torch.FloatTensor`` of shape ``(batch_size, hidden_size)``
+            Last layer hidden-state of the first token of the sequence (classification token)
+            further processed by a Linear layer. This output is to be used to embed questions for
+            nearest neighbors queries with context embeddings.
+
+    Examples::
+
+        tokenizer = DprTokenizer.from_pretrained('dpr-base-uncased')
+        model = DprQuestionEncoder.from_pretrained('dpr-ctx_encoder-base')
+        input_ids = torch.tensor(tokenizer.encode("Hello, is my dog cute ?")).unsqueeze(0)  # Batch size 1
+        embeddings = model(input_ids)  # the embeddings of the given question.
+
+    """
+
+    def __init__(self, config: DprConfig):
+        super().__init__(config)
+        self.config = config
+        self.question_encoder = get_bert_question_encoder_component(config)
+        self.init_weights()
+
+    def forward(self, input_ids: T, token_type_ids: Optional[T] = None, attention_mask: Optional[T] = None) -> T:
+        if attention_mask is None:
+            attention_mask = input_ids != self.config.pad_id
+            attention_mask = attention_mask.to(device=input_ids.device)
+        sequence_output, pooled_output, hidden_states = self.question_encoder(
+            input_ids, token_type_ids, attention_mask
+        )
+        return pooled_output
+
+
+DprSpanPrediction = collections.namedtuple(
```

Comment:
Awesome to use named tuples here :-) In Reformer, I named all named tuples `Ouput` in the end to make sure one does not confuse a named tuple with a Layer => Think this would help here as well

Reply:
What about `DPRReaderOutput` ?